lib/api/objectCopy.js

```diff
@@ -98,6 +105,25 @@ function _prepMetadata(sourceObjMD, headers, sourceIsDestination, authInfo,
     const locationMatch = headers[locationHeader] ?
         sourceObjMD[locationHeader] === headers[locationHeader] : true;
 
+    const locationTypeMatch = headers[locationHeader] ?
+    config.locationConstraints[sourceObjMD[locationHeader]].type ===
+    config.locationConstraints[headers[locationHeader]].type : true;
+
+    let sourceExternalBucketName;
+    if (sourceObjMD[locationHeader] &&
+      config.locationConstraints[sourceObjMD[locationHeader]].type
+      === 'aws_s3') {
+        sourceExternalBucketName =
```

Comment:
we should add a method to the Config class that is getExternalBucketName() so that you don't have to have all of this logic here and it will handle the different layouts of the azure and aws configs.

Reply:
I put this check directly to external backend. It's simpler and clearer.